res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,774 @@
+PioneerDDJ400.loopout = [false, false]; // out loop is pressed
+PioneerDDJ400.loopAdjustMultiply = 5;
+
+PioneerDDJ400.samplerCallbacks = [];
+
+// Wrapper to easily ignore the function when the button is released.
+var ignoreRelease = function(fn) {
+    "use strict";
+    return function(channel, control, value, status, group) {
+        if (value === 0) { // This means the button is released.
+            return;
+        }
+        return fn(channel, control, value, status, group);
+    };
+};
+
+
+PioneerDDJ400.init = function() {
+    "use strict";
+    // init controller
+    // init tempo Range to 10% (default = 6%)
+    engine.setValue("[Channel1]", "rateRange", PioneerDDJ400.tempoRanges[1]);
+    engine.setValue("[Channel2]", "rateRange", PioneerDDJ400.tempoRanges[1]);
+
+    // show focus buttons on Effect Ract 1 only
+    engine.setValue("[EffectRack1_EffectUnit1]", "show_focus", 1);
+    engine.setValue("[EffectRack1_EffectUnit2]", "show_focus", 0);
+    engine.setValue("[EffectRack1_EffectUnit3]", "show_focus", 0);
+
+    // Connect the VU-Meter LEDS
+    engine.connectControl("[Channel1]", "VuMeter", "PioneerDDJ400.vuMeterUpdate");
+    engine.connectControl("[Channel2]", "VuMeter", "PioneerDDJ400.vuMeterUpdate");
+
+    // reset vumeter
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.deck1.vuMeter, false);
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.deck2.vuMeter, false);
+    
+    // DJ3730:  added
+    // enable soft takeover for rate controls
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+
+
+    // DJ3730:  added
+    // Sampler callbacks
+    for (i = 1; i <= 16; ++i) {
+        PioneerDDJ400.samplerCallbacks.push(engine.makeConnection("[Sampler" + i + "]", "play", PioneerDDJ400.samplerPlayOutputCallbackFunction));
+    }
+
+   // DJ3730:  added
+   // trigger "track loaded" animations when a track is loaded
+   for (i=1; i<=2; i++) {
+        engine.connectControl("[Channel"+i+"]","track_loaded", "PioneerDDJ400.trackLoadedLED"+i);
+        engine.trigger("[Channel"+i+"]","track_loaded");
+    }
+
+    // DJ3730:  added
+    // eye candy : play the "track loaded" animation on both decks at startup
+    midi.sendShortMsg(0x9F,0x00,0x7F);
+    midi.sendShortMsg(0x9F,0x01,0x7F);
+
+    // poll the controller for current control positions on startup
+    midi.sendSysexMsg([0xF0,0x00,0x40,0x05,0x00,0x00,0x02,0x06,0x00,0x03,0x01,0xf7], 12);
+};
+
+PioneerDDJ400.trackLoadedLED1 = function (loaded) {
+   PioneerDDJ400.trackLoadedLED(1, loaded);
+}
+PioneerDDJ400.trackLoadedLED2 = function (loaded) {
+   PioneerDDJ400.trackLoadedLED(2, loaded);
+}
+
+PioneerDDJ400.trackLoadedLED = function (channel, loaded) {
+   if (loaded) {
+      var value = 0x7F;
+   } else {
+      var value = 0x00;
+   }
+   midi.sendShortMsg(0x9F,0x00+(channel-1),value);
+}
+
+PioneerDDJ400.toggleLight = function(midiIn, active) {
+    "use strict";
```

Comment:
This doesn't actually do anything in Mixxx's ancient JS interpreter, so it is misleading to have it. Please remove it. Even if it did work, adding it to every function but not the global context creates a mess.

When we merge #1795 for Mixxx 2.4, ES7 will be supported. In ES6, strict mode is automatically on in many contexts like classes and modules.

Reply:
These have been present in the code for several months now (going back to at least December) but it's no problem for me to remove them.

Reply:
Done